spring-rabbit/src/test/java/org/springframework/amqp/rabbit/config/QueueParserIntegrationTests.java

```diff
@@ -74,7 +74,7 @@ public void testArgumentsQueue() throws Exception {
 		assertEquals(null, result);
 
 		connectionFactory.destroy();
-
+		brokerIsRunning.getAdmin().deleteQueue("arguments");
```

Comment:
The foo queue is never used; it can be removed; as I mentioned in the other comments, it doesn't really help to set auto-delete because it causes tests to fail against a broker that has the older tests run against it.